### PR TITLE
Adds initial representation of the data returned in records

### PR DESCRIPTION
**Pull request by JPrevost**
#### What does this PR do?

Adds an initial representation of the data we'll be returning to end users of the API.

#### Helpful background context

Additional work on ontologies and normalization of data is ongoing and will result in changes to this representation of the data.

#### How can a reviewer manually see the effects of these changes?

I've placed a Swagger view of the API here to view for your convenience:
https://app.swaggerhub.com/apis/JPrevost/mit-libraries_discovery_api/0.0.2

#### What are the relevant tickets?

- https://mitlibraries.atlassian.net/browse/DIP-116

#### Requires Full Reindexing of all Sources?
NO

#### Includes new or updated dependencies?
NO

### CODE REVIEW

File: openapi.yml [hunk 1]
```diff
@@ -55,16 +93,80 @@ components:
       properties:
         id:
           type: string
+        type:
```
Comment (gravesm) [inline openapi.yml]: `type` may be valid as a property name, but it is a little confusing given that it is used elsewhere in the spec to mean something else. Can we come up with an alternative property name?

File: openapi.yml [hunk 2]
```diff
@@ -46,6 +46,44 @@ components:
...
+          type: array
+          items:
+            type: string
+        issns:
+          type: array
+          items:
+            type: string
+        doi:
+          type: string
+    Link:
```
Comment (gravesm) [inline openapi.yml]: I think we should consider what we want out of links. Are we just looking for a bag o' undifferentiated links? Are we looking for machine actionable links? There may be existing classifications of links we could use like the rel types for microformats or something. What we want to do with them should probably drive how we model them.

Comment (JPrevost) [reply openapi.yml]: Yes, exactly the discussion I am looking forward to having. I'm not sure we'll have a decision in the short term so maybe we can open a ticket to revisit the Link model and schedule a time to discuss it.

Comment (gravesm) [reply openapi.yml]: Yeah, I think it's fine to just merge as is and we can adjust later.

Comment (hakbailey) [reply openapi.yml]: The ontology includes defined terms for some links (such as [http://id.loc.gov/ontologies/bibframe.html#p_electronicLocator](http://id.loc.gov/ontologies/bibframe.html#p_electronicLocator) for a link to the item itself), but other links will be harder to classify and may just require a local controlled vocabulary

Comment (JPrevost) [reply openapi.yml]: I'm adding that to the ticket I opened to discuss this further:
https://mitlibraries.atlassian.net/browse/DIP-119

File: openapi.yml [hunk 3]
```diff
@@ -55,16 +93,81 @@ components:
       properties:
         id:
           type: string
+        category:
```
Comment (hakbailey) [inline openapi.yml]: I would maybe distinguish here between content type and physical format, e.g.:
```
content_type:
    description: High level categorization of the type of content, such as text, still image, audio, etc.
    type: string
format:
    description: High level categorization of the content format, such as online resource, CD, book, etc.
    type: string
```
Outstanding question about how specific we want to get with content types...

File: openapi.yml [hunk 4]
```diff
@@ -46,6 +46,44 @@ components:
...
+        url:
+          type: string
+          format: uri
+        restrictions:
+          type: string
+        kind:
+          description: what kind of link this is, such as full_text, restricted_full_text, sfx, unknown
+          type: string
     Record:
       description: An abbreviated representation of a record
```
Comment (hakbailey) [inline openapi.yml]: What determines which fields make it into the abbreviated record? Just wondering if that's something we want to discuss further.

File: openapi.yml [hunk 5]
```diff
@@ -55,16 +93,81 @@ components:
...
+          format: uri
+        year:
+          type: string
         title:
           type: string
+        full_text_links:
+          type: array
+          items:
+            $ref: '#/components/schemas/Link'
+        authors:
```
Comment (hakbailey) [inline openapi.yml]: Added ticket to discuss and decide on author/creator/contributor models with metadata folks [https://mitlibraries.atlassian.net/browse/DIP-120](https://mitlibraries.atlassian.net/browse/DIP-120)

File: openapi.yml (hunk 3 above)
Comment (JPrevost) [reply openapi.yml]: I'll make that change if that's what the ontologists are going with for sure. Thanks.

Comment (hakbailey) [reply openapi.yml]: It's standard practice in all metadata schemas, even super high level ones like Dublin Core. But we can wait to make the change until we've decided exactly which terms to use.

File: openapi.yml (hunk 4 above)
Comment (JPrevost) [reply openapi.yml]: tldr; Bento uses what is in the abbreviated record in the initial search results and the full record for the, er, full record view.

We could eliminate that distinction and always return full records in the search results view as well as the full record view (we still need both API end points, but they could return the same models if we aren't concerned with performance which honestly now that you called this out I suspect would be negligible).

File: openapi.yml (hunk 3 above)
Comment (JPrevost) [reply openapi.yml]: Nah, I'll make your suggested changes for now and we can tweak if we need to later. 👍

**Pull request merged**